sourced/ml/algorithms/uast_struct_to_bag.py

```diff
@@ -28,44 +36,49 @@ def _uast2seq(self, root, walk):
                 nodes[id(child)].extend(child.children)
                 stack.append(child)
             else:
-                walk.append(stack.pop())
+                sequence.append(stack.pop())
+        return sequence
 
-    def uast_to_bag(self, uast):
+    def __call__(self, uast):
         bag = defaultdict(int)
-        seq = []
-        self._uast2seq(uast, seq)
+        node_sequence = self._uast2sequence(uast)
 
         # convert to str - requirement from wmhash.BagsExtractor
-        seq = "".join([self.node2ind(n) for n in seq])
+        node_sequence = "".join([self.node2index[n] for n in node_sequence])
 
-        if isinstance(self.seq_len, int):
```

Comment:
this part was moved here https://github.com/src-d/ml/pull/145/files#diff-86f78a3692ae357bb5a7628fcf0cbc6bR11 up to a parent class.